tofhir-server/src/main/scala/io/tofhir/server/service/TerminologySystemService.scala

```diff
@@ -0,0 +1,137 @@
+    val result: Future[Unit] = terminologySystemRepository.deleteTerminologySystem(id)
+
+    result.map{ terminologySystem =>
+      // update Jobs Terminology fields if it is successful
+      updateJobTerminology(id)
+      terminologySystem
+    }
+  }
+
+  /**
+   * Check whether job's id and terminologySystem id is equal
+   * @param job job object to be checked
+   * @param terminologySystemId id of terminogySystem
+   * @return Boolean indicating whether job's id and terminologySystem's id are equal
+   */
+  private def checkEqualityOfIds(job: FhirMappingJob, terminologySystemId: String): Boolean = {
+    // Get terminology service of the job
+    val terminologyServiceSettings: LocalFhirTerminologyServiceSettings =
+      job.terminologyServiceSettings.get.asInstanceOf[LocalFhirTerminologyServiceSettings]
+
+    terminologyServiceSettings.folderPath.split('/').lastOption.get.equals(terminologySystemId)
+  }
+
+  /**
+   * Update Job's terminology fields with updated terminology system
+   * @param id id of terminologySystem
+   * @param terminologySystem updated terminology system
+   * @return
+   */
+  private def updateJobTerminology(id: String, terminologySystem: Option[TerminologySystem] = Option.empty): Future[Unit] = {
+    // Get project ids from job cache
+    val projectIds: Seq[String] = mappingJobRepository.getCachedMappingsJobs.keys.toSeq
+
+    projectIds.foreach { projectId =>
+      // Get jobs of given project
+      val jobs: Future[Seq[FhirMappingJob]] = mappingJobRepository.getAllJobs(projectId)
+
+      jobs.map { jobs =>
+        jobs
+          .filter { job =>
+            job.terminologyServiceSettings.isDefined
+          }
+          .filter { job =>
+            job.terminologyServiceSettings.get.isInstanceOf[LocalFhirTerminologyServiceSettings]
+          }
+          .filter { job =>
+            checkEqualityOfIds(job, id) // Skip if id is not equal
+          }
+          .foreach { job =>
+            // Create updated job object
+            val updatedJob = terminologySystem match {
+              // Update terminology service case
+              case Some(ts) => {
+                val terminologyServiceSettings: LocalFhirTerminologyServiceSettings =
+                  job.terminologyServiceSettings.get.asInstanceOf[LocalFhirTerminologyServiceSettings]
+
+                val updatedTerminologyServiceSettings: LocalFhirTerminologyServiceSettings =
+                  terminologyServiceSettings.copy(conceptMapFiles = ts.conceptMaps, codeSystemFiles = ts.codeSystems)
+
+                job.copy(terminologyServiceSettings = Some(updatedTerminologyServiceSettings))
+              }
+              // Delete terminology service case
+              case None => job.copy(terminologyServiceSettings = None)
+            }
+            // Update job in the repository
+            mappingJobRepository.putJob(projectId, job.id, updatedJob)
+          }
+      }
+    }
+    Future.apply(None)
```

Comment:
Why is this here? Instead, you can wrap whole method inside a Future { }

Reply:
A new Future is created for each projectId within foreach loop. Those Futures should be handled appropriately.